README.md

```diff
@@ -34,14 +43,143 @@ then `func` can discover it and expose it to a user in a similar way the static
   project creation (e.g. group and artifact ids for maven project).
 * The `executable` behaviour may be customized by environment variables.
 
+The advantage of the `Contract A` is simpler implementation,
+but it would be usable mostly just for `func`.
+
+## Executable Contract B
+
+* The `executable` shall support three sub-commands: `runtimes`, `templates` and `create`.
+* The `runtimes` sub-command shall take no parameters.</br>
+  The sub-command shall print a JSON array of string to standard output containing a list of supported runtimes.
+* The `templates` sub-command shall take exactly one parameter: runtime name. </br>
+  The sub-command shall print a JSON Object where keys are names of templates and values are specification
+  of questions/parameters needed by a template.</br>
+  Example of such a JSON: 
+  ```json
+  {
+    "my-java-template": {
+      "questions": [
+        {
+          "name": "artifact-id",
+          "type": "string"
+        },
+        {
+          "name": "group-id",
+          "type": "string"
+        },
+        {
+          "name": "build-system",
+          "type": "select",
+          "items": [
+            "maven",
+            "gradle"
+          ]
+        },
+        {
+          "name": "hypothetical",
+          "type": "string",
+          "depends-on": "build-system",
+          "ask-when": {
+            "condition": {
+              "type": "equals",
+              "rhs": ".build-system",
+              "lhs": "'maven'"
+            }
+          }
+        }
+      ]
+    }
+  }
+  ```
+* The `create` sub-command shall accept four argument:
+  * --name: name of the Function project
+  * --language: language of the Function project
+  * --template: the template to create the Function project</br>
+  The sub-command shall use `stdio` in a specific way:</br></br>
+  On `stdin` there would be additional parameters provided as an JSON Object based on specification provided
+  by the `templates` sub-command for given language/template.</br>
+  Example of such a JSON:
+  ```json
+  {
+    "answers": [
+      {
+        "name": "artifact-id",
+        "value": "my-art"
+      },
+      {
+        "name": "group-id",
+        "value": "my.grp"
+      },
+      {
+        "name": "build-system",
+        "value": "gradle"
+      }
+    ]
+  }
+  ```
+  The sub-command is supposed to return project files as `tar` archive on `stdout`.
+
+The advantage of the `Contract B` is that it could be better for external tools,
+but implementation would be more difficult.
+
+## Containerization
```

Comment:
What is your opinion on this?
How much we want limit the `executable`?

Reply:
I am pretty torn on this. Arbitrary code execution on a developer's computer is pretty sketchy. But a container is quite limiting. I'm leaning towards accepting those limits, however, and saying we should just keep it to containers. This also (potentially) simplifies implementation of the repo/template binary for providers because they will always be able to control the execution environment.